k8s/client.go

```diff
@@ -448,3 +456,206 @@ func (client *Client) FindDeploymentByName(ctx context.Context, name string) (*a
+		if err != nil {
+			fmt.Printf("Failed to apply manifest: %v", err)
+			return nil, err
+		}
+
+		appliedSuccessfully = append(appliedSuccessfully, []string{created.GetKind(), created.GetName()})
+	}
+	return appliedSuccessfully, nil
+}
+
+//func rollbackManifests(ctx context.Context, config *restclient.Config, manifests [][]string) error {
+//	if len(manifests) == 0 {
+//		return nil
+//	}
+//	fmt.Printf("An error ocurred while applying manifests\n")
+//	fmt.Printf("Trying to rollback %d manifests\n", len(manifests))
+//
+//	dynamicClient, err := dynamic.NewForConfig(config)
+//	if err != nil {
+//		return err
+//	}
+//	fmt.Printf("Trying to rollback %d manifests\n", len(manifests))
+//	for _, manifest := range manifests {
+//		manifest = strings.TrimSpace(manifest)
+//		if manifest == "" {
+//			continue
+//		}
+//
+//		decoder := yamlk8s.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
+//		obj := &unstructured.Unstructured{}
+//		_, gvk, err := decoder.Decode([]byte(manifest), nil, obj)
+//		if err != nil {
+//			fmt.Printf("Failed to decode manifest: %v", err)
+//			continue
+//		}
+//
+//		kindPluralized := strings.ToLower(gvk.Kind) + "s"
+//		resource := dynamicClient.Resource(gvk.GroupVersion().WithResource(kindPluralized))
+//
+//		err = resource.Namespace(obj.GetNamespace()).Delete(ctx, obj.GetName(), metav1.DeleteOptions{})
+//		if err != nil {
+//			fmt.Printf("Failed to delete manifest: %v", err)
+//			return err
+//		}
+//
+//		fmt.Printf("Deleted %s %s\n", obj.GetKind(), obj.GetName())
+//	}
+//	return nil
+//}
+
+func splitManifest(manifest []byte) []string {
+	return strings.Split(string(manifest), "---\n")
+}
+
+func getOperatorManifest(version string) ([]byte, error) {
+	const operatorReleases = "https://api.github.com/repos/calyptia/core-operator-releases/releases"
+
+	type Release struct {
+		TagName string `json:"tag_name"`
+		Assets  []struct {
+			BrowserDownloadUrl string `json:"browser_download_url"`
+		} `json:"assets"`
+	}
+
+	resp, err := http.Get(operatorReleases)
+	if err != nil {
+		return nil, err
+	}
+	defer func(Body io.ReadCloser) {
+		err := Body.Close()
+		if err != nil {
+			fmt.Println("Error closing response body:", err)
+		}
+	}(resp.Body)
+
+	var releases []Release
+	err = json.NewDecoder(resp.Body).Decode(&releases)
+	if err != nil {
+		return nil, err
+	}
+
+	if version == "" {
+		version = "v1.0.0-alpha0"
+	}
+
+	var urlForDownload string
+	for _, release := range releases {
+		if release.TagName == version {
+			urlForDownload = release.Assets[0].BrowserDownloadUrl
+		}
+	}
+
+	response, err := http.Get(urlForDownload)
+	if err != nil {
+		fmt.Println("Error downloading manifest:", err)
+		return nil, err
+	}
+	defer func(Body io.ReadCloser) {
+		err := Body.Close()
+		if err != nil {
+			fmt.Println("Error closing response body:", err)
+		}
+	}(response.Body)
+
+	manifestBytes, err := io.ReadAll(response.Body)
+	if err != nil {
+		fmt.Println("Error reading manifest:", err)
```

Comment:
do we we want to println at this level or just return and print at the first level?